docs/_docs/04_community/client-factory.md

```diff
@@ -0,0 +1,124 @@
+
+To configure different cloud environments, we will use the same `AZURE_ENVIRONMENT` environment variable that the Go SDK uses. This can currently be set to one of the following values:
+
+|Value                      |Cloud Environment  |
+|---------------------------|-------------------|
+|"AzureChinaCloud"          |ChinaCloud         |
+|"AzureGermanCloud"         |GermanCloud        |
+|"AzurePublicCloud"         |PublicCloud        |
+|"AzureUSGovernmentCloud"   |USGovernmentCloud  |
+|"AzureStackCloud"          |Azure stack        |
+
+When using the "AzureStackCloud" setting, you MUST also set the `AZURE_ENVIRONMENT_FILEPATH` variable to point to a JSON file containing your Azure Stack URI details.
+
+## Putting it all together
+
+ `client_factory` implements this pattern described above in order to instantiate and return properly configured *REST SDK for GO* clients so that test implementers don't have to consider REST API client implementation as long as they have the correct `AZURE_ENVIRONMENT` env setting.  If this environment variable is not set, the client will assume public cloud as the cloud environment to communicate with.  We strongly recommend developers creating Terratest helper methods for Azure use this pattern with client factory to create REST API clients.  This will reduce effort for Terratest users creating test for Azure resources.
+
+>NOTE: TERRAFORM uses [ARM_ENVIRONMENT](https://www.terraform.io/docs/backends/types/azurerm.html#environment) environment variable to set the correct cloud environment.  
+<!-- -->
+>NOTE: The default behavior of the `client_factory` is to use the AzurePublicCloud environment. This requires no work from the developer to configure, and ensures consistent behavior with the current SDK code.
+
+### Wait, I don't see the client in client factory for the rest api I want to interact with
+
+ If you require a client that is not already implemented in client factory for your helper method, you will need to create a corresponding method that instantiates the client and accepts base URI following the patterns discussed.  Below is a walkthrough for adding a client to client factory.
+
+## Walkthrough, adding a client to client_factory
+
+### Add your client namespace to client factory
+
+In the Azure SDK for GO, each service should have a module that implements that services client.  You can find the correct module [here](https://godoc.org/github.com/Azure/azure-sdk-for-go).  Add that module to the client factory imports.  Below is an example for client imports that shows clients for compute, container service and subscriptions.
+
+{% include examples/explorer.html example_id='client-factory' file_id='client_factory_code' class='wide quick-start-examples' skip_learn_more=true skip_view_on_github=true skip_tags=true snippet_id='client_factory_example.imports' %}
```

Comment:
A few of these seem broken:

<img width="851" alt="Screen Shot 2020-11-17 at 5 21 17 PM" src="https://user-images.githubusercontent.com/711908/99424347-7cc22c80-28f9-11eb-8f32-a1b9968deac1.png">

But some do work, so I don't think it's a general problem with the snippets, but perhaps just a couple broken IDs or links or typos?

<img width="874" alt="Screen Shot 2020-11-17 at 5 21 22 PM" src="https://user-images.githubusercontent.com/711908/99424310-72079780-28f9-11eb-9392-7b3d277a4a9a.png">

Reply:
The 404's are a result of the content being pulled from the "master" branch of the live repo, which doesn't yet contain the appropriate files (they're in the PR). Also, when an entire file is loaded, it means the tags/snippets weren't found and the `examples.js` logic falls back to rendering the entire file.